iodata/formats/mwfn.py

```diff
@@ -0,0 +1,367 @@
+    (7, 'p'): HORTON2_CONVENTIONS[(7, 'p')],
+    (6, 'p'): HORTON2_CONVENTIONS[(6, 'p')],
+    (5, 'p'): HORTON2_CONVENTIONS[(5, 'p')],
+    (4, 'p'): HORTON2_CONVENTIONS[(4, 'p')],
+    (3, 'p'): HORTON2_CONVENTIONS[(3, 'p')],
+    (2, 'p'): HORTON2_CONVENTIONS[(2, 'p')],
+    (0, 'c'): ['1'],
+    (1, 'c'): ['x', 'y', 'z'],
+    (2, 'c'): ['xx', 'yy', 'zz', 'xy', 'xz', 'yz'],
+    (3, 'c'): ['xxx', 'yyy', 'zzz', 'xyy', 'xxy', 'xxz', 'xzz', 'yzz', 'yyz', 'xyz'],
+    (4, 'c'): HORTON2_CONVENTIONS[(4, 'c')][::-1],
+    (5, 'c'): HORTON2_CONVENTIONS[(5, 'c')][::-1],
+    (6, 'c'): HORTON2_CONVENTIONS[(6, 'c')][::-1],
+    (7, 'c'): HORTON2_CONVENTIONS[(7, 'c')][::-1],
+    (8, 'c'): HORTON2_CONVENTIONS[(8, 'c')][::-1],
+    (9, 'c'): HORTON2_CONVENTIONS[(9, 'c')][::-1],
+}
+
+
+def _load_helper_opener(lit: LineIterator, keys: list) -> Tuple[int, float, float,
+                                                                float, float, float, int]:
+    """Read initial variables."""
+    max_count = len(keys)
+    count = 0
+    d = {}
+    while count < max_count:
+        line = next(lit)
+        for name in keys:
+            if name in line:
+                d[name] = line.split('=')[1].strip()
+                count += 1
+    return int(d['Wfntype']), float(d['Charge']), float(d['Naelec']), float(d['Nbelec']),\
+        float(d['E_tot']), float(d['VT_ratio']), int(d['Ncenter'])
+
+
+def _load_helper_basis(lit: LineIterator) -> Tuple[int, int, int, int, int]:
+    """Read initial variables."""
+    # Nprims must be last or else it gets read in with Nprimshell
+    basis_keywords = ["Nbasis", "Nindbasis", "Nshell", "Nprimshell", "Nprims", ]
+    max_count = len(basis_keywords)
+    count = 0
+    d = {}
+    next(lit)
+    while count < max_count:
+        line = next(lit)
+        for name in basis_keywords:
+            if name in line:
+                d[name] = int(line.split('=')[1].strip())
+                count += 1
+                break
+    return d['Nbasis'], d['Nindbasis'], d['Nprims'], d['Nshell'], d['Nprimshell']
+
+
+def _load_helper_atoms(lit: LineIterator, num_atoms: int) -> Tuple[np.ndarray, np.ndarray,
+                                                                   np.ndarray]:
+    """Read the coordinates of the atoms."""
+    atnums = np.empty(num_atoms, int)
+    atcorenums = np.empty(num_atoms, float)
+    atcoords = np.empty((num_atoms, 3), float)
+    line = next(lit)
+    while '$Centers' not in line and line is not None:
+        line = next(lit)
+
+    for atom in range(num_atoms):
+        line = next(lit)
+        atnums[atom] = int(line.split()[2].strip())
+        atcorenums[atom] = float(line.split()[3].strip())
+        # extract atomic coordinates
+        coords = line.split()
+        atcoords[atom, :] = [coords[4], coords[5], coords[6]]
+        # return but convert angstroms to amu
+    return atnums, atcorenums, atcoords / spc.value(u'atomic unit of length') / 1E10
```

Comment:
yes, replacing with atcoords * angstroms. I would like to note that I find this confusing however. It would be nicer if iodata.utils defined this as ang_to_amu or something that is explicitly a conversion from one unit to another. While one knows angstroms are involved in atcoords * angstoms, it is not clear what the conversion unit to/from is.

Reply:
Given that iodata explicitly states it uses atomic units, perhaps my above comment if a bit narrow viewed, and upon remembering that iodata specifically has default units, all conversions must be to/from them.

Reply:
Indeed, the `ang_to_bohr` style would be ok if the "to" part were not be fixed by convention.

The following makes it easy to remember how it works:

- When you say "this bond length is 1.5 Å", the python equivalent is `bond_length = 1.5 * angstrom`.

- The other way is often found in graphs in old papers, where the axis labels to indicate a bond length in angstrom is printed as "Bond length / Å". Expressing this in python would be `print("Bond length in Angstrom:", bond_length / angstrom)`

Should we add this explanation to the docs somewhere? It might be helpful.

Reply:
It can't hurt although I was more after the fact that only one unit is known. But, I had to correct myself because IOData always uses bohr for its length units, so if you see a conversion of lengths, you know it is going from Bohr to whatever units is given, e.g., angstrom.

Reply:
ok, thanks.